packages/react-reconciler/src/__tests__/useEffectEvent-test.js

```diff
@@ -742,52 +742,57 @@ describe('useEffectEvent', () => {
       return <Text text={`Welcome to the ${roomId} room!`} />;
     }
 
-    act(() => ReactNoop.render(<ChatRoom roomId="general" theme="light" />));
-    assertLog(['Welcome to the general room!']);
+    await act(async () =>
+      ReactNoop.render(<ChatRoom roomId="general" theme="light" />),
+    );
+    await act(async () => jest.runAllTimers());
+    assertLog(['Welcome to the general room!', 'Connected! theme: light']);
     expect(ReactNoop).toMatchRenderedOutput(
       <span prop="Welcome to the general room!" />,
     );
 
-    jest.advanceTimersByTime(100);
-    Scheduler.unstable_advanceTime(100);
-    assertLog(['Connected! theme: light']);
-
     // change roomId only
-    act(() => ReactNoop.render(<ChatRoom roomId="music" theme="light" />));
-    assertLog(['Welcome to the music room!']);
+    await act(async () =>
+      ReactNoop.render(<ChatRoom roomId="music" theme="light" />),
+    );
+    await act(async () => jest.runAllTimers());
+    assertLog([
+      'Welcome to the music room!',
+      // should trigger a reconnect
+      'Connected! theme: light',
+    ]);
+
```

Comment:
Is there any reason these are combined now?

Reply:
Our internal version of act calls `runPendingTimers` as a way to force Suspense fallbacks to show up. Unlike the public `act` API which instead pushes things to an internal queue. We do it that way because our tests need to run in both prod and dev, whereas the public `act` only works in dev.

For that reason it's really best to not use setTimeout in our tests, unless you make sure to get the "native" ones instead of Jest's fake ones. But this test was already written that way.

In this case it's not important to the test's behavior so it's fine to flush them all together. I only kept the `jest.runAllTimers()` calls in case we reimplement our internal `act` at some point to not run timers internally. It passes without those.